portal-impl/src/com/liferay/portal/events/LoginPostAction.java

```diff
@@ -106,12 +106,13 @@ public void run(
 					userId = PortalUtil.getUserId(httpServletRequest);
 				}
 
-				boolean reindex = false;
-
-				if (UserLocalServiceUtil.addDefaultGroups(userId) ||
-					UserLocalServiceUtil.addDefaultRoles(userId) ||
-					UserLocalServiceUtil.addDefaultUserGroups(userId)) {
+				boolean reindex = UserLocalServiceUtil.addDefaultGroups(userId);
 
+				if (UserLocalServiceUtil.addDefaultRoles(userId)) {
+					UserLocalServiceUtil.addDefaultUserGroups(userId);
+					reindex = true;
+				}
+				else if (UserLocalServiceUtil.addDefaultUserGroups(userId)) {
```

Comment:
probably you can write something like 
```
boolean reindex = false;

if (UserLocalServiceUtil.addDefaultGroups(userId) |
					UserLocalServiceUtil.addDefaultRoles(userId) |
					UserLocalServiceUtil.addDefaultUserGroups(userId)) {
  reindex = true;					
}
```

so basically is the same as the original but with a single pipe |

Reply:
Got it, thanks!